resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientResponse.java

```diff
@@ -70,12 +71,38 @@ public void setClientConfiguration(ClientConfiguration configuration)
       this.processor = configuration;
    }
 
-   @Override
-   public Object getEntity()
-   {
-      abortIfClosed();
-      return super.getEntity();
-   }
+	@Override
+	public Object getEntity() {
+		abortIfClosed();
+		Object entity = super.getEntity();
+		if (entity != null) {
+			return entity;
+		}
+		//Check if the entity was previously fully consumed
+		try {
+			InputStream inputStream = getEntityStream();

```

Comment:
The call to getEntityStream() here is possibly a problem. Basically, we need to make sure that method is not called while a ClientResponse object is garbage collected as a consequence of finalize method having been aggressively called. So far, we didn't bother about this problem because the only invoker to that protected method is already synchronized (readFrom(...) ). In any case, it would probably be safe to simply turn getEntityStream() into a synchronized method.